[Harith Bin Nurhisham] iP

harithhh06 opened the pull request:
### LunaBot

> You can only manage time if you track it right. - Spica's team [(source)](https://www.spica.com/blog/the-best-time-management-quotes)

Why remember tasks when a bot can do it for you?

Bogged down with too many submissions, events and things to do? Use LunaBot to manage your tasks!

LunaBot is:

- _text-based_
- easy to use
- **SUPER FAST**  🚀
- Updated in real time

All you need to do is,

1. download it from [here](https://github.com/Harithhh06/ip/releases).
2. Start the app
3. add your tasks ✅
4. let it manage your tasks for you 📋
5. And it's **FREE**?!?! 🤯

Features:

- [x] Managing tasks
- [ ] Managing deadlines
- [ ] Scheduling events

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class LunaBot {
    public static void main(String[] args) {
        String filepath;
        new LunaBot(filepath).run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

harithhh06 committed:
Level 0. Rename, Greet, Exit

harithhh06 committed:
Level 1. Echo

harithhh06 committed:
Level 2. Add, List

harithhh06 committed:
Added task class

harithhh06 committed:
Revert "Added task class"

This reverts commit 8164ba69c816f0c02b2b33d1ac22cc3824d7a60c.

harithhh06 committed:
Level-3: Mark as Done

harithhh06 committed:
Level-3: Mark as Done

harithhh06 committed:
Level 4. ToDos, Events, Deadlines

harithhh06 committed:
Level 5. Handle Errors

harithhh06 committed:
Level 6. Delete

harithhh06 committed:
A-TextUiTesting: Automated Text UI Testing

harithhh06 committed:
Level 7: Save

harithhh06 committed:
add txt file

harithhh06 committed:
Level-8: Dates and Times

nigellqx commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,78 @@
+import java.io.*;
```
Perhaps listing out all the imports might be better according to the coding standards?

nigellqx commented on src/main/java/LunaBot.java:
```diff
@@ -0,0 +1,245 @@
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class LunaBot {
```
Perhaps have a Ui and a Parser class to handle different tasks?

nigellqx commented on src/main/java/LunaBot.java:
```diff
@@ -0,0 +1,245 @@
+        while (true) {
+            try {
+                input = scanner.nextLine();
+                if (input.equals("bye")) {
```
Perhaps you can use input.equalsIgnoreCase("bye") to handle cases where user might input "BYE"?

nigellqx replied:
minor nitpick - maybe the use of a switch statement is better for readability?

nigellqx commented on src/main/java/LunaBot.java:
```diff
@@ -0,0 +1,245 @@
+    private static void markTask(String input) throws LunaBotException {
+        try {
+            String taskNumString = input.substring(5);
+            // Checks to see if user provided a number
```
I like how you add comments in the nested code to allow readers to know what you are doing at each stage of the code

nigellqx commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,78 @@
+        this.filePath = filePath;
+    }
+
+    public ArrayList<Task> load() throws LunaBotException {
```
minor nitpick - perhaps there could be comments in this method to guide the readers?

nigellqx reviewed (commented):
LGTM - just some minor considerations regarding the code quality

harithhh06 committed:
A-MoreOOP

harithhh06 committed:
A-packages

harithhh06 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

harithhh06 committed:
A-Gradle

harithhh06 committed:
A-JUnit

harithhh06 committed:
A-Jar

harithhh06 committed:
A-JavaDoc

harithhh06 committed:
fix event task formatting

harithhh06 committed:
A-CodingStandard

harithhh06 committed:
Resolved merge conflicts

harithhh06 committed:
Fixed issues with command class and naming of filed

harithhh06 committed:
fixed issues again

harithhh06 committed:
Level-9 Find

harithhh06 committed:
Level-10: GUI

harithhh06 committed:
A-Assertions

Implement Assertions for Input and State Validation

Add assertions across various methods to validate inputs and state
conditions. This improves reliability and debugging by enforcing
expected conditions during development.

harithhh06 committed:
Merge pull request #2 from Harithhh06/A-Assertions

A-Assertions

harithhh06 committed:
A-CodeQuality

Refactor code to follow Code Quality as stated in the course website

Add guard clauses to classes accordingly.

harithhh06 committed:
Merge pull request #3 from Harithhh06/A-CodeQuality

A-CodeQuality

harithhh06 committed:
C-FriendlierSyntax
Make syntax friendlier for users

Shorten commands words for some commands

harithhh06 committed:
Merge pull request #4 from Harithhh06/C-FriendlierSyntax

C-FriendlierSyntax

harithhh06 committed:
C-Help

Add Help command to print a list of available commands to the user.
Display welcome message when app is started which informs user of the
help command.

harithhh06 committed:
Merge pull request #5 from Harithhh06/C-Help

C-Help

harithhh06 committed:
A-Personality
Change responses of LunaBot from default Duke responses.
Give LunaBot more personality

harithhh06 committed:
Add Ui.png file to docs
Delete all data from tasks.txt

harithhh06 committed:
A-UserGuide

Add product description and user guide in the README.md file

harithhh06 committed:
Merge pull request #6 from Harithhh06/A-UserGuide

A-UserGuide

harithhh06 committed:
ensure program properly exits on command
